plugin/pkg/shoot/quotavalidator/admission_test.go

```diff
@@ -266,6 +276,34 @@ var _ = Describe("quotavalidator", func() {
 				err := admissionHandler.Validate(attrs, nil)
 				Expect(err).NotTo(HaveOccurred())
 			})
+
+			Context("machine type in cloud profile defines stoage", func() {
+				It("should pass because quota is large enough", func() {
+					shoot2 := *shoot.DeepCopy()
+					shoot2.Spec.Provider.Workers[0].Machine.Type = machineTypeName2
+					shoot2.Spec.Provider.Workers[0].Volume.Size = "19Gi"
+
+					quotaProject.Spec.Metrics[garden.QuotaMetricStorageStandard] = resource.MustParse("20Gi")
+
+					attrs := admission.NewAttributesRecord(&shoot2, nil, garden.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, garden.Resource("shoots").WithVersion("version"), "", admission.Create, false, nil)
+
+					err := admissionHandler.Validate(attrs, nil)
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				It("should fail because quota is not large enough", func() {
+					shoot2 := *shoot.DeepCopy()
+					shoot2.Spec.Provider.Workers[0].Machine.Type = machineTypeName2
+					shoot2.Spec.Provider.Workers[0].Volume.Size = "21Gi"
+
+					quotaProject.Spec.Metrics[garden.QuotaMetricStorageStandard] = resource.MustParse("20Gi")
+
+					attrs := admission.NewAttributesRecord(&shoot2, nil, garden.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, garden.Resource("shoots").WithVersion("version"), "", admission.Create, false, nil)
+
+					err := admissionHandler.Validate(attrs, nil)
+					Expect(err).To(HaveOccurred())
```

Comment:
```suggestion
					Expect(err).To(HaveOccurred())
					Expect(err.Error()).To(ContainSubstring("Quota limits exceeded"))
```

I think, it should be checked, that the returned error is really the expected one.